### PR TITLE
Chat mention alignment fix

### LINKED ISSUE #15693: Mentions in chat are not aligned properly with rest of the text
Design: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=3188-50051&t=n9pKlzSeSiKJFayB-4

### Screenshots
Android | iOS
-|-
Android|iOS

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #15693 

### Screenshots
Android | iOS
-|-
Android | iOS

### Affected areas
- 1:1 chat
- Community chat

### Known issue
- Border radius not working

status: ready

### CODE REVIEW

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (ajayesivan) [inline src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: This component is inside the chat `Text` component. While all other inline elements inside the chat `Text` component are of type `Text`, here we are using the `Touchable` component, and it was causing alignment issues.

To fix this issue, I replaced the Touchable with a Text component. However, since this new `Text` component is nested inside another `Text` component, it won't support `padding` and `border-radius`. To create a similar effect to `padding`, I added a single space character on each side. But, with this implementation `border-radius` is still an issue.

I would appreciate your thoughts and suggestions on this.

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (J-Son89) [inline src/status_im2/contexts/chat/messages/content/text/view.cljs]: consider making a helper function outside this scope and leave the comment there.
e.g 

```
 ;; Text inside Text doesn't support padding, so we need to add a space on each side to mimic
;; that effect
(defn pad-text [s]
(str " " s " ")
```

**Review by J-Son89 — APPROVED** [review]
I like the conversation in the description image 😆😄

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: @ajayesivan, my first thoughts were that we're trading one bug for another, so I'm not sure if it's worth the trouble. Have you thought about using the `Pressable` API instead of a `Touchable`? Do you think that could lead to better results and no alignment issues?

**Review by flexsurfer — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (ajayesivan) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: It's not a perfect solution. But I think it's better than this:

Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: Indeed, it looks terrible. Are you going to create a separate follow-up issue for the new "bugs" being introduced by this PR?

**Review by ilmotta — APPROVED** [review]

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (ajayesivan) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: I didn't try `Pressable` but I think it won't make any difference(I tried replacing the touchable with `View`). From my understanding, the root cause of this issue is that each nested `Text` component is not using Flexbox. When we use a `View` or any other element besides Text, it uses Flexbox, which can cause misalignment issues.

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (ajayesivan) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs]: I'm not sure if we will need to reuse this. However, if you think it's best to create a helper function for this, is there a common place we keep this kind of helpers?

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (ajayesivan) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: > Indeed, it looks terrible. Are you going to create a separate follow-up issue for the new "bugs" being introduced by this PR?

I will do that 👍.

Comment (ajayesivan) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: https://github.com/status-im/status-mobile/issues/15702

Comment (ilmotta) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs:43]: Thank you @ajayesivan!

File: src/status_im2/contexts/chat/messages/content/text/view.cljs
Comment (J-Son89) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs]: Just in this file, to keep it separate. Not particularly a big fan of big comments within a block of code. Moving it to the helper function means the name explains what it is doing and then the comment is there beside the helper function is someone wants to get more info

Comment (ajayesivan) [reply src/status_im2/contexts/chat/messages/content/text/view.cljs]: Updated as per your suggestion.

**Review by briansztamfater — APPROVED** [review]
LGTM :)

**Comment by pavloburykh** [comment]
@ajayesivan thanx for the PR. Tested and ready to be merged.

**Comment by ajayesivan** [comment]
https://github.com/status-im/status-mobile/pull/15723 - introduces a complete fix. We should wait until #15723 to get merged and then we can close this PR without merging.

**Comment by ajayesivan** [comment]
Closing this PR since the original issue has been solved by another PR:https://github.com/status-im/status-mobile/pull/15723

**Pull request closed without merging**
